### PR TITLE
feat: keep unset component value

### PR DESCRIPTION
**Pull request by Mark-Greenbook**
Read from component input and update values user specify only. And a little bit SRP change. #17

### CODE REVIEW

**Comment by Zyin055** [comment]
This almost works, but it doesn't work for dropdowns. 

For example, when removing the `txt2img_sampling` value from the config file, this happens:
```
Config Presets: changed to 12
Components before: {'txt2img_batch_size': 1, 'txt2img_cfg_scale': 7, 'txt2img_denoising_strength': 0.3, 'txt2img_sampling': 15, 'txt2img_hr_scale': 2, 'txt2img_enable_hr': False, 'txt2img_batch_count': 1, 'txt2img_steps': 10, 'txt2img_width': 704, 'txt2img_height': 960}
Components after : {'txt2img_batch_size': 1, 'txt2img_cfg_scale': 7, 'txt2img_denoising_strength': 0.3, 'txt2img_sampling': 'DPM++ 2M Karras', 'txt2img_hr_scale': 2, 'txt2img_enable_hr': False, 'txt2img_batch_count': 1, 'txt2img_steps': 12, 'txt2img_width': 704, 'txt2img_height': 960}
```
The before value is `15`, and the after value is `DPM++ 2M Karras`. This causes the dropdown to not be set to any value, and causes an error during image generation:

```
Traceback (most recent call last):
  File "C:\Stuff\AI\Stable Diffusion\modules\call_queue.py", line 56, in f
    res = list(func(*args, **kwargs))
  File "C:\Stuff\AI\Stable Diffusion\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "C:\Stuff\AI\Stable Diffusion\modules\txt2img.py", line 28, in txt2img
    sampler_name=sd_samplers.samplers[sampler_index].name,
TypeError: list indices must be integers or slices, not NoneType
```

File: scripts/config_presets.py [hunk 1]
```diff
@@ -280,37 +287,25 @@ def after_component(self, component, **kwargs):
...
-                                    )
+                            # update component values with user preset
+                            current_components = dict(zip(component_map.keys(),components_value))
+                            print("Components before:", current_components)
+                            current_components.update(config_preset)
 
+                            # transform necessary components from index to value
+                            for k,v in current_components.items():
+                                if k in index_type_components and type(v) == int:
+                                    current_components[k] = component_map[k].choices[v]
```
Comment (Mark-Greenbook) [inline scripts/config_presets.py]: Gradio index component get in as int and need a string output. Weird, but this should work now.

**Comment by Zyin055** [comment]
Looks better now. Was there anything else you wanted to put in?

**Comment by Mark-Greenbook** [comment]
> Looks better now. Was there anything else you wanted to put in?

That's it. Nice extension btw.

**Comment by Mark-Greenbook** [comment]
> Looks better now. Was there anything else you wanted to put in?

That's it. Nice extension btw.

**Pull request merged**